internal/store/store_test.go

```diff
@@ -1,15 +1,188 @@
 package store
 
 import (
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
 	"testing"
 )
 
+//These are global variables that are used across more than one test function
+var Dir, _ = ioutil.TempDir("/tmp", "osbuild-composer-test-")
+var myStore = New(&Dir)
+var myCustomizations = blueprint.Customizations{}
+var myBP = blueprint.Blueprint{
+	Name:        "testBP",
+	Description: "Testing blueprint",
+	Version:     "0.0.1",
+	Packages: []blueprint.Package{
+		{Name: "test1", Version: "*"}},
+	Modules: []blueprint.Package{
+		{Name: "test2", Version: "*"}},
+	Groups: []blueprint.Group{
+		{Name: "test3"}},
+	Customizations: &myCustomizations,
+}
+var myBP2 = blueprint.Blueprint{
+	Name:        "WIPtestBP",
+	Description: "Testing blueprint 2",
+	Version:     "0.0.1",
+	Packages: []blueprint.Package{
+		{Name: "test1", Version: "2"}},
+	Modules: []blueprint.Package{
+		{Name: "test2", Version: "3"}},
+	Groups: []blueprint.Group{
+		{Name: "test3"}},
+	Customizations: &myCustomizations,
+}
+var myBP3 = blueprint.Blueprint{
+	Name:        "testBP",
+	Description: "Newer Testing blueprint",
+	Version:     "0.0.2",
+	Packages: []blueprint.Package{
+		{Name: "test1", Version: "2"}},
+	Modules: []blueprint.Package{
+		{Name: "test2", Version: "3"}},
+	Groups: []blueprint.Group{
+		{Name: "test3"}},
+	Customizations: &myCustomizations,
+}
+
+//This variable is for storing a commit hash
+var CommitHash string
+var CommitHashLatest string
+
+func TestMain(m *testing.M) {
+	//run all the tests
+	exitVal := m.Run()
+	//remove tmp directory
+	os.RemoveAll(Dir)
```

Comment:
Clean-ups should be always in `defer` because resources will leak if `m.Run` panics.

Reply:
os.Exit doesn't respect defer

Reply:
Right, I forgot about that. Hmm, that makes `TestMain` with a resource cleanup kinda hard to write.

Reply:
The approach taken here looks correct to me. First run the tests, then cleanup and then exit. Same thing happens in internal/client/integration_test.go